### PR TITLE
hoc2015: Custom collision regions for variety of background/map combos.

### PR DESCRIPTION
**Pull request by breville**
Custom collision regions added for each background's horizontal, circle, and blobs layouts.

Note that these rectangles have been carefully chosen to stay within the bounds of the default collision rectangles that are based on tiles.  Some logic, such as moving a character to a free square if a map is changed, relies on the default collision rectangles, so these new rectangles must always be smaller.

Just for fun, I got to use my Narrativ.io project to draw the rectangles.

screenshot 2015-11-20 12 22 32

### CODE REVIEW

**Comment by breville** [comment]
fyi @cpirich

**Comment by islemaster** [comment]
Nice illustration of the change!

**Comment by islemaster** [comment]
What's the big white square in the middle of the top-right (snow+blobs) layout?

File: apps/src/studio/skins.js [hunk 1]
```diff
@@ -608,6 +608,86 @@ function loadHoc2015(skin, assetUrl) {
     'rebelpilot':      { frames: 13, width: 100, height: 100, scale: 1,   renderOffset: { x: 0, y: -25}, activity: 'flee',  speed: constants.SpriteSpeed.SLOW, spritesCounterclockwise: true },
   };
 
+  skin.customObstacleZones = {endor : {}, hoth: {}, starship: {}};
+
+  skin.customObstacleZones.endor.blobs = [
+    { minx: 361, maxx: 413, miny: 261, maxy: 344 },
```
Comment (islemaster) [inline apps/src/studio/skins.js]: nit: Why aren't these capitalized `minX` `maxX` `minY` `maxY`?

As-is, I did a double-take.
image

File: apps/src/studio/studio.js [hunk 2]
```diff
@@ -1510,20 +1510,54 @@ Studio.willCollidableTouchWall = function (collidable, xCenter, yCenter) {
...
+                          rectCenterX,
+                          rectWidth / 2 + collidableWidth / 2,
+                          yCenter,
+                          rectCenterY,
+                          rectHeight / 2 + collidableHeight / 2)) {
+        return true;
+      }
+    }
+  } else {
+    // Compare against regular wall tiles.
```
Comment (islemaster) [inline apps/src/studio/studio.js:1540]: This is good in a "not breaking anything" way but eventually I'd like to see these two cases unified a bit - seems like in the non-`customObstacleZones` case we should just push the regular wall tiles on as collisionRects and go through the same checks, right?

**Comment by islemaster** [comment]
LGTM

File: apps/src/studio/skins.js (hunk 1 above)
Comment (breville) [inline apps/src/studio/skins.js]: Fixed.

**Comment by breville** [comment]
The big white square is the main sprite's collision box; being fixed up in separate PRs as it happens.

**Pull request merged**